### PR TITLE
Add disableAnalysis flag to layerProps

### PR DESCRIPTION
**Pull request by reed1**
This fixes issue #643.

How to test the feature:

- Set "run_analysis' to false in layers.json, for example:
image

- The above will hide the layer in Analysis hazard layer selection
image

@wadhwamatic, I'm using "run_analysis" prop, but since boolean flag commonly defaults to false, show we use "hide_in_analysis" instead?

### CODE REVIEW

**Comment by ericboucher** [comment]
Indeed, I think the flag should be reversed. Maye something like `disableAnalysis`? @wadhwamatic  what do you think?

File: frontend/src/components/MapView/Layers/LayerDropdown.tsx [hunk 1]
```diff
@@ -95,12 +95,20 @@ function LayerDropdown({
...
-        layers: category.layers.filter(layer =>
-          layer.type === 'wms'
-            ? layer.type === type &&
-              [undefined, 'polygon'].includes(layer.geometry)
-            : layer.type === type,
-        ),
+        layers: category.layers.filter(layer => layer.type === type),
+      }))
+      // 4. additional filter for WMS layers
+      .map(category => ({
```
Comment (ericboucher) [inline frontend/src/components/MapView/Layers/LayerDropdown.tsx:101]: I would suggest creating a function `filterLayersForAnalysis` that explicitly details the steps we use to filter out layers.

This way you can call it in the `.map` above and don't need to loop through the list one more time with a no-op for most categories / layers

**Comment by wadhwamatic** [comment]
> Indeed, I think the flag should be reversed. Maye something like `disableAnalysis`? @wadhwamatic what do you think?

Yes, that sounds good. Please make that change @reed1

**Comment by reed1** [comment]
Thanks for the suggestion. I've updated the PR. @wadhwamatic @ericboucher

File: frontend/src/components/MapView/Layers/LayerDropdown.tsx [hunk 2]
```diff
@@ -67,6 +67,16 @@ function LayerDropdown({
     tables: [],
   };
 
+  const filterLayersForAnalysis = (layer: LayerType) => {
+    if (layer.type === 'wms') {
+      return (
+        !layer.disableAnalysis &&
```
Comment (ericboucher) [inline frontend/src/components/MapView/Layers/LayerDropdown.tsx:73]: By design, I think layer.disableAnalysis should affect all layers, not just wms layers.

File: frontend/src/components/MapView/Layers/LayerDropdown.tsx [hunk 3]
```diff
@@ -67,6 +67,16 @@ function LayerDropdown({
     tables: [],
   };
 
+  const filterLayersForAnalysis = (layer: LayerType) => {
```
Comment (ericboucher) [inline frontend/src/components/MapView/Layers/LayerDropdown.tsx:70]: Let's add some comments in this function please

File: frontend/src/components/MapView/Layers/LayerDropdown.tsx [hunk 4]
```diff
@@ -67,6 +67,17 @@ function LayerDropdown({
     tables: [],
   };
 
+  // Filter out layers that are not supported by the analysis tool
+  const filterLayersForAnalysis = (layer: LayerType) => {
+    if (layer.disableAnalysis) {
+      return false;
+    }
+    if (layer.type === 'wms') {
+      return [undefined, 'polygon'].includes(layer.geometry);
```
Comment (ericboucher) [inline frontend/src/components/MapView/Layers/LayerDropdown.tsx:76]: can you explicit what type of geometry is not supported?

Comment (ericboucher) [reply frontend/src/components/MapView/Layers/LayerDropdown.tsx:76]: (point and linestring)

**Review by ericboucher — APPROVED** [review]

**Pull request merged**